packages/flame/test/components/component_test.dart

```diff
@@ -200,6 +200,56 @@ void main() {
       );
     });
   });
+
+  group('Component descendants', () {
+    flameGame.test(
+      'length must be equal to the number of added components',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+        await game.ensureAdd(component);
+
+        expect(game.descendants().length, 3);
+      },
+    );
+
+    flameGame.test(
+      'length must be equal to the number of added components including itself',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+        await game.ensureAdd(component);
+
+        expect(game.descendants(includeSelf: true).length, 4);
+      },
+    );
+
+    flameGame.test(
+      'length must be 0 if [hasPendingLifecycleEvents] is true',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+        game.add(component);
+
+        expect(game.hasPendingLifecycleEvents, true);
+        expect(game.descendants().length, 0);
+      },
+    );
+
+    flameGame.test(
+      'length should not change '
+      'if [hasPendingLifecycleEvents] is true after adding',
+      (game) async {
+        final component = Component()..add(Component()..add(Component()));
+        await game.add(component);
```

Comment:
```suggestion
        await game.ensureAdd(component);
```

For testing, I prefer `ensureAdd` over `add`, for consistency. Same applies for other.

Reply:
`game.add()` is perfectly fine here, and it's closer to the "real" code since there is no `ensureAdd()` in production.

Reply:
Just out of curiosity, when would you `ensureAdd` over `add`?